### DISCUSSION TITLE
v5: ScrollSpy issue when loading page with hash

### DISCUSSION THREAD

**Discussion by smolenski-mikolaj** [twbs/bootstrap#31569] [General]
After loading a page containing a hash with a reference to a specific section, **scrollSpy** reads the wrong window offsetTop and marks the wrong link as active. Then, after the manual scroll it still sets wrong links as active. 

A not very efficient workaround is to prevent the default window.scroll after the page is loaded and manually init another window.scroll. 

**Here's the code:**
```
  if (window.location.hash) {
    const target = document.getElementById(window.location.hash.substring(1));

    setTimeout(function() {
      window.scroll({
        top: 0
      });
      setTimeout(() => {
        window.scroll({
          top: target.offsetTop - 100,
          behavior: 'smooth'
        });
      }, 1000);
    }, 1);
  }
```

Unfortunately, the refresh method, delayed init or full re-init does not work in this case. 
Is there any better solution to this problem?

Thanks in advance for your reply.

---
Bootstrap Version: 5.0.0-alpha1
Operating system: Windows 10
Browser: Firefox 80, Chrome 84

**Comment by aribudin** [comment]
I have same problem
You have solution ?

**Reply by bavarianbytes** [reply]
Same here.
